config/timezones.json

```diff
@@ -0,0 +1,596 @@
+[
```

Comment:
What happens when this file does not exist?  Does it fall back to code?

We have a lot of automated installs that may need to change if this file is required.

Reply:
There is no fallback.  If the file does not exist, the server cannot run without it.

It would be a similar issue if the `config.json` file did not exist.  Unless I'm missing something, I believe there's no fallback if `config.json` is missing. 

Quick test: move `config,default.json` into a `config/tmp` directory and run the server.  I get this error:

```
Error: LoadConfig: Error opening config file=config.json, err=no config file found,
[15:27:50 EDT 2018/03/19] [INFO] (github.com/mattermost/mattermost-server/app.(*App).Shutdown:206) Stopping Server...
[15:27:50 EDT 2018/03/19] [INFO] (github.com/mattermost/mattermost-server/app.(*App).HubStop:114) stopping websocket hub connections
[15:27:50 EDT 2018/03/19] [WARN] (github.com/mattermost/mattermost-server/app.(*App).HubStop:119) We appear to have already sent the stop checking for deadlocks command
[15:27:50 EDT 2018/03/19] [INFO] (github.com/mattermost/mattermost-server/app.(*App).Shutdown:225) Server stopped
[15:27:50 EDT 2018/03/19] [CRIT] (github.com/mattermost/mattermost-server/cmd/commands.runServer:64) LoadConfig: Error opening config file=config.json, err=no config file found,
```



> We have a lot of automated installs that may need to change if this file is required.

I'm curious what that setup looks like.   The `timezones.json` file should be packaged with the automated install script, is this handled by MM or is this a user-specific feature?

Is this what you are referring to? https://docs.mattermost.com/guides/orchestration.html?highlight=automated%20installs#deployment-solution-program-requirements


Let me know how I can help with this issue if there's anything I can do 😄 

Thanks for the catch @coreyhulen

Reply:
@coreyhulen is there a way I can test automated installation?

Reply:
Yeah not sure it's comparable to config.json since that's integral to the system running.

Why not have it in code?  I doubt this file would ever change.  Or if you really want the file then have both.  So it will fallback to the code version when the file is missing similar to localization.

Not sure you can test all the various installers.  For example the one I pretty sure will fail would be Gitlab Onimbus integration.  They generate the config directory and file at a different location.

Reply:
I agree it's not integral to the system running.  For some context, this is a discussion I had with @hmhealey earlier last week in this PR. 

> I actually just realized a potential problem here. What happens if the user specifies the path to their config.json using an environment variable and we try to load the timezones.json? This affects any GitLab Omnibus installs in particular since they like to store their files all over the place.
>
> It might be a bit hokey, but we might need to add a config setting for the path to the timezones file so that it can be loaded properly. Something like TimezoneSettings.AvailableTimezonesPath that, if empty, defaults to doing what it does currently?

@coreyhulen I made a change to support GitLab Omnibus users.  Would this suffice?

Reply:
I left it in a json file in order for admins to configure if they desire.
Say they want to only support timezones in one region.

@dmeza had a suggestion that I believe would work fine.
Have a `default_timezones.json` and `timezones.json` file. Similar to `default and config.json`.
Would that work?
If `timezones.json` does not exists we recreate it from `default_timezones.json`

Reply:
I'm not certain either one of those works 100%.  `TimezoneSettings.AvailableTimezonesPath` helps but the gitlab installer still needs to change.  default_timezones.json still has the same problem (it would need to exist in the dir).  Unless you meant `default_timezones.json` is in a go string that is deserialized when the proper file is missing.

Reply:
> Unless you meant default_timezones.json is in a go string that is deserialized when the proper file is missing.

The idea would be the same as `config/default.json` and `config.json`.
If `config.json` does not exists, fallback to `default.json`.

I was not thinking about a go string per se.

I understand your concern and I feel it's unnecessary to update the automated install process.
I'll add some logic to default to a list of supported timezones in go code.  That should be a succinct solution.

Reply:
:+1: